### PR TITLE
Update Registry Manager to support multi-tenancy and OpenSearch Serverless

### LINKED ISSUE #66: Update to utilize new multi-tenancy approach
## 💡 Description

* handshake auth to only access data owned by the node
* load-dd function only for EN
* update documentation for load-dd to open ticket with EN

### PR DESCRIPTION
**Pull request by al-niessner**
## 🗒️ Summary
Adapt to new registry-common

## ⚙️ Test Data and/or Report
TBD

## ♻️ Related Issues
Closes #66

### CODE REVIEW

**Comment by tloubrieu-jpl** [comment]
Hi @al-niessner , the branch `issue_66` here is meant to work with branch `issue_36` of registry-common ?

If yes, the pom.xml in this repository should reference dependency on registry-common 1.6.0-SNAPSHOT, but it does not. Can you help me with that ? 

Thanks

**Comment by al-niessner** [comment]
> Hi @al-niessner , the branch `issue_66` here is meant to work with branch `issue_36` of registry-common ?
> 
> If yes, the pom.xml in this repository should reference dependency on registry-common 1.6.0-SNAPSHOT, but it does not. Can you help me with that ?
> 
> Thanks

Yes these 3 PRs work together:
NASA-PDS/registry-common#47
NASA-PDS/registry-mgr#73
NASA-PDS/harvest#152

**Comment by tloubrieu-jpl** [comment]
Hi Al,

I build registry-common from branch ` issue_36`
I switched the dependency on registry-common to 1.6.0-SNAPSHOT in the pom.xml here.
I tried to build registry-mgr, and I am getting these errors:
```
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /Users/loubrieu/git/registry-mgr/src/main/java/gov/nasa/pds/registry/mgr/dao/DataExporter.java:[12,46] package gov.nasa.pds.registry.common.es.client does not exist
[ERROR] /Users/loubrieu/git/registry-mgr/src/main/java/gov/nasa/pds/registry/mgr/dao/DataExporter.java:[13,46] package gov.nasa.pds.registry.common.es.client does not exist
[ERROR] /Users/loubrieu/git/registry-mgr/src/main/java/gov/nasa/pds/registry/mgr/dao/DataExporter.java:[14,46] package gov.nasa.pds.registry.common.es.client does not exist
[ERROR] /Users/loubrieu/git/registry-mgr/src/main/java/gov/nasa/pds/registry/mgr/cmd/data/UpdateAltIdsCmd.java:[19,40] package gov.nasa.pds.registry.common.cfg does not exist
[ERROR] /Users/loubrieu/git/registry-mgr/src/main/java/gov/nasa/pds/registry/mgr/cmd/data/UpdateAltIdsCmd.java:[20,46] package gov.nasa.pds.registry.common.es.client does not exist
[ERROR] /Users/loubrieu/git/registry-mgr/src/main/java/gov/nasa/pds/registry/mgr/cmd/dd/DeleteDDCmd.java:[16,46] package gov.nasa.pds.registry.common.es.client does not exist
[ERROR] /Users/loubrieu/git/registry-mgr/src/main/java/gov/nasa/pds/registry/mgr/cmd/dd/DeleteDDCmd.java:[17,46] package gov.nasa.pds.registry.common.es.client does not exist
[ERROR] /Users/loubrieu/git/registry-mgr/src/main/java/gov/nasa/pds/registry/mgr/cmd/dd/LoadDDCmd.java:[8,40] package gov.nasa.pds.registry.common.cfg does not exist
[ERROR] /Users/loubrieu/git/registry-mgr/src/main/java/gov/nasa/pds/registry/mgr/cmd/dd/LoadDDCmd.java:[30,13] cannot find symbol
  symbol:   class RegistryCfg
  location: class gov.nasa.pds.registry.mgr.cmd.dd.LoadDDCmd
[ERROR] /Users/loubrieu/git/registry-mgr/src/main/java/gov/nasa/pds/registry/mgr/cmd/data/DeleteDataCmd.java:[16,46] package gov.nasa.pds.registry.common.es.client does not exist
[ERROR] /Users/loubrieu/git/registry-mgr/src/main/java/gov/nasa/pds/registry/mgr/cmd/data/DeleteDataCmd.java:[17,46] package gov.nasa.pds.registry.common.es.client does not exist
[ERROR] /Users/loubrieu/git/registry-mgr/src/main/java/gov/nasa/pds/registry/mgr/dao/RegistryManager.java:[7,40] package gov.nasa.pds.registry.common.cfg does not exist
[ERROR] /Users/loubrieu/git/registry-mgr/src/main/java/gov/nasa/pds/registry/mgr/dao/RegistryManager.java:[33,29] cannot find symbol
  symbol:   class RegistryCfg
  location: class gov.nasa.pds.registry.mgr.dao.RegistryManager
[ERROR] /Users/loubrieu/git/registry-mgr/src/main/java/gov/nasa/pds/registry/mgr/dao/RegistryManager.java:[60,29] cannot find symbol
  symbol:   class RegistryCfg
  location: class gov.nasa.pds.registry.mgr.dao.RegistryManager
[ERROR] /Users/loubrieu/git/registry-mgr/src/main/java/gov/nasa/pds/registry/mgr/util/es/EsDocWriter.java:[12,46] package gov.nasa.pds.registry.common.es.client does not exist
[ERROR] /Users/loubrieu/git/registry-mgr/src/main/java/gov/nasa/pds/registry/mgr/util/es/EsDocWriter.java:[30,68] package SearchResponseParser does not exist
[ERROR] /Users/loubrieu/git/registry-mgr/src/main/java/gov/nasa/pds/registry/mgr/cmd/dd/ListDDCmd.java:[8,40] package gov.nasa.pds.registry.common.cfg does not exist
[ERROR] /Users/loubrieu/git/registry-mgr/src/main/java/gov/nasa/pds/registry/mgr/cmd/data/ExportFileCmd.java:[11,46] package gov.nasa.pds.registry.common.es.client does not exist
[ERROR] /Users/loubrieu/git/registry-mgr/src/main/java/gov/nasa/pds/registry/mgr/cmd/data/ExportFileCmd.java:[12,46] package gov.nasa.pds.registry.common.es.client does not exist
[ERROR] /Users/loubrieu/git/registry-mgr/src/main/java/gov/nasa/pds/registry/mgr/cmd/data/ExportFileCmd.java:[13,46] package gov.nasa.pds.registry.common.es.client does not exist
[ERROR] /Users/loubrieu/git/registry-mgr/src/main/java/gov/nasa/pds/registry/mgr/cmd/data/ExportFileCmd.java:[33,68] package SearchResponseParser does not exist
[INFO] 21 errors 

```

I am not finding the classes missing in the branch issue_36 of registry-common. 

Thanks

**Comment by al-niessner** [comment]
Is 1.6.0-SNAPSHOT deployed?

**Comment by tloubrieu-jpl** [comment]
Hi @al-niessner ,

I am building the registry-common locally and it is deployed locally in my .m2 repository.

**Comment by tloubrieu-jpl** [comment]
Hi @al-niessner ,

I tried the set-archive-status subcommand but that did not work, see my terminal here with 2 attempts:

```
% ./bin/registry-manager set-archive-status -index jaxa-registry -auth /Users/loubrieu/Documents/pds/registry/es-auth.txt -es file:/Users/loubrieu/Documents/pds/registry/mcp_dev.xml -lidvid urn:nasa:pds:insight_rad:data_derived::7.0 -status archived 
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
[INFO] Setting product status. LIDVID = urn:nasa:pds:insight_rad:data_derived::7.0, status = archived
[ERROR] Request failed: [http_exception] server returned 404
loubrieu@MT-212318 registry-manager-4.7.0-SNAPSHOT % ./bin/registry-manager set-archive-status -auth /Users/loubrieu/Documents/pds/registry/es-auth.txt -es file:/Users/loubrieu/Documents/pds/registry/mcp_dev.xml -lidvid urn:nasa:pds:insight_rad:data_derived::7.0 -status archived 

SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
[INFO] Setting product status. LIDVID = urn:nasa:pds:insight_rad:data_derived::7.0, status = archived
[ERROR] Request failed: [http_exception] server returned 404
loubrieu@MT-212318 registry-manager-4.7.0-SNAPSHOT % 
```

Before I merge the PR, can you also resolve the current conflict with the main branch ?

Thanks,

Thomas

**Comment by al-niessner** [comment]
@tloubrieu-jpl 

How did you resolve conflict from main branch for the status subject?

**Comment by tloubrieu-jpl** [comment]
Hi @al-niessner ,

I don't understand your question.

I did not resolve the conflict but I was asking if you could do it, by rebasing your development branch on main. And as an independent request from me, I was also asking if you could fix the error when using the set-archive-status subcommand.

Let me know what I missed.

Thanks

**Comment by al-niessner** [comment]
@tloubrieu-jpl 

Fix for status is waiting for you in PR NASA-PDS/registry-common#49. Merge is complete.

**Review by tloubrieu-jpl — APPROVED** [review]
I am approving although there are remaining know bugs.

File: src/main/java/gov/nasa/pds/registry/mgr/cmd/reg/KnownRegistryCmd.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package gov.nasa.pds.registry.mgr.cmd.reg;
+
+import java.net.URL;
+import org.apache.commons.cli.CommandLine;
+import gov.nasa.pds.registry.common.connection.KnownRegistryConnections;
+import gov.nasa.pds.registry.mgr.cmd.CliCommand;
+
+public class KnownRegistryCmd implements CliCommand {
+  public void run(CommandLine cmdLine) throws Exception {
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply src/main/java/gov/nasa/pds/registry/mgr/cmd/reg/KnownRegistryCmd.java:10]: fixed

**Pull request merged**